### ISSUE TITLE
Sugestion: Add option loop - auto replay channel

### ISSUE THREAD

**Issue by rapi3** [kodi-pvr/pvr.iptvsimple#382]
Hi,
Any chance to add a loop option - auto replay in case stream have errors and play stop and exit to channel list?
thank you

**Comment by phunkyfish** [comment]
I know this seems like a simple request but it's actually very complex.

Have you tried the Matrix version of the addon first? There is some logic in there to help with dropped connections.

**Comment by rapi3** [comment]
No I did not try Matrix as I don't like installing others addons on my rpi especially if are not from official repo.
I only have this addon and youtube.

**Comment by phunkyfish** [comment]
Matrix (Kodi 19) is official repo too 😉

Kodi 18 is bug fixes only, no new features there so any implementation would be kodi 19 only anyway.

**Comment by rapi3** [comment]
I have latest stable LibreElec on rpi so whatever version it is there I am using now.

**Comment by phunkyfish** [comment]
If you want to try the matrix version there are official test builds here: https://forum.kodi.tv/showthread.php?tid=343068

Just backup your settings so you can go back to the old version after trying it. If it doesn’t help please get a debug log while on it if you decide to try it. Who knows, it may solve all your issues.

**Comment by phunkyfish** [comment]
Did you try Matrix? Did it improve?

**Comment by rapi3** [comment]
Not yet as it is still beta on LibreElec.

**Comment by daniol** [comment]
So, after 2 years, is there any option to force repeat a stream in case of an error, until is manually stopped?
Is there any option in Kodi? Any option that can be supplied in the m3u?
Where to implement this in code?

**Comment by phunkyfish** [comment]
No, there is not. Some like this would need to be added in the input-streams i’s say. So that they detect the end of the stream and start it again. Or somehow support playlists in PVR.
